### ISSUE TITLE
Python and Vim highlighters are broken

### ISSUE THREAD

**Issue by sudo-julia** [nvim-treesitter/nvim-treesitter#1976]
### Before reporting

- [X] I have updated my neovim version to latest _master_
- [X] I have updated my plugin to the latest version
- [X] I have run `:TSUpdate`

### Describe the bug

Python and Vim highlighting fail because of the errors:

```
## The following errors have been detected:
  - ERROR: python(highlights): /usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 4695
  - ERROR: vim(highlights): /usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 787
```

Disabling highlighting for these filetypes does not even temporarily fix the issue, as the error messages persist. It makes it hard to write code when error messages will randomly capture a key 

### To Reproduce

1. Compile and install neovim from master branch
2. Install tree-sitter
3. Install nvim-treesitter with packer.nvim
4. Install Python and Vim parsers with `:TSInstall python vim`

### Expected behavior

The highlighters should work.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found  0.20.0 (67de9435b109f9bd8bf5957d11ff13161966d262) (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v16.11.1 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (GCC) 11.1.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - vim            x ✓ . . ✓ 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - python         x ✓ ✓ ✓ ✓ 
  - bash           ✓ ✓ ✓ . ✓ 
  - json           ✓ ✓ ✓ ✓ . 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

## The following errors have been detected:
  - ERROR: vim(highlights): /usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 787
  - ERROR: python(highlights): /usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 4695
```

### Output of `nvim --version`

```text
NVIM v0.6.0-dev+554-g2481b18a5
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/jam/build/neovim/build/config -I/home/jam/build/neovim/src -I/home/jam/build/neovim/.deps/usr/include -I/usr/include -I/home/jam/build/neovim/build/src/nvim/auto -I/home/jam/build/neovim/build/include
Compiled by sudo-julia@hostname

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info
```

### Additional context

I'm running Arch Linux with the 5.14.15-arch1-1 kernel, x86_64.

The Vim highlighter has been broken for a long time, but the Python highlighter broke sometime last week and has heavily disrupted my programming workflow. Error messages make programming in a Python buffer near undoable, even with all treesitter features disabled through `:TSBufDisable highlight`. If there's a way to stop these messages from occurring, even with the highlighter broken, please let me know.

Also, I'm aware of #1887 and the fix that was found yesterday. However, there are no duplicate folders for me to remove. [here](https://github.com/nvim-treesitter/nvim-treesitter/issues/1887#issuecomment-958177253) is my comment from that thread.

So far, I've tried:
- Installing neovim from master
- Installing treesitter from master
- Uninstalling and reinstalling the broken parsers
- Compiling treesitter parsers with gcc and clang
- Reinstalling nvim-treesitter with packer.nvim
- Removing the folder at ~/.local/share/nvim/site/pack/packer/start/nvim-treesitter and reinstalling
- Removing the folder at ~/.local/share/nvim/site/pack/packer/start/ and reinstalling all plugins

**Comment by theHamsta** [comment]
Are you using the bundled dependencies or did you compile Tree-sitter by yourself (either by cloning the repo or via brew).

Can you check with github.com/nvim-treesitter/playground (using the query linter) which nodes are invalid?

**Comment by sudo-julia** [comment]
Tree-sitter was compiled myself with the [AUR package](https://aur.archlinux.org/packages/tree-sitter-git/). The version is `tree-sitter 0.20.0 (67de9435b109f9bd8bf5957d11ff13161966d262)`.

After installing nvim-treesitter/playground, this is the new error message upon opening a pyhton file:  
```text
Error executing vim.schedule lua callback: [string "<nvim>"]:66: /usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 4695
stack traceback:
        [C]: in function '(for generator)'
        [string "<nvim>"]:66: in function <[string "<nvim>"]:61>
```

If you need something more specific, please let me know!

**Comment by theHamsta** [comment]
We're testing with a tree-sitter version from 30.September (the official revision in Neovim's thrid party cmake) while yours is from October 25th. It's possible that there were changes that introduces an incompatibility. Can you please use the query linter of the playground to check which nodes cause problems?

**Comment by sudo-julia** [comment]
I'm going to be completely honest, I have 0 idea how Treesitter works so it's very difficult for me to parse the issues that're flooding the screen. I've also read the README of nvim-treesitter/playground and I'm still not too sure how to get you the information you're asking for.
![This](https://user-images.githubusercontent.com/62857870/141202678-f7a3f3a9-2d06-4d3e-9ed9-221e4a9196f0.png) is what floods the screen after enabling the query linter and opening a python file. If this isn't what you're looking for then please let me know because I want to help out, I just don't know how.

**Comment by sudo-julia** [comment]
In the meantime I'd like to uninstall the python parser, but even after running `TSUninstall python`, it's listed as installed and still floods the screen with error messages upon opening a python file. Is this a known issue?

**Comment by theHamsta** [comment]
It's possible that the python parser has been installed by to another location and this is most likely the reason for your issue (that nvim-treesitter cannot update the python parser). Can you the to use `:h nvim_get_runtime`  to find `parser/python.so`.

I meant using the query linter to analyze what's wrong in `queries/python/highlights.scm`

**Comment by tropicaltidbits** [comment]
I am getting the same errors in Python files immediately after updating from Neovim 0.5.1 to current master: `query: invalid node type at position 4695`. It doesn't matter what type of node is actually typed into the buffer.

**Comment by meridionaljet** [comment]
I am getting the same errors in Python files immediately after updating from Neovim 0.5.1 to current master: `query: invalid node type at position 4695`. It doesn't matter what type of node is actually typed into the buffer.

**Comment by meridionaljet** [comment]
This is the offending commit:

```
git show 10419ee747512a5189a6755925a456025228435b
commit 10419ee747512a5189a6755925a456025228435b (HEAD)
Author: Santos Gallegos <stsewd@protonmail.com>
Date:   Thu Sep 30 21:06:29 2021 -0500

    Python: highlight interpolation escaped chars

    Support for these was just added in the grammar.

diff --git a/queries/python/highlights.scm b/queries/python/highlights.scm
index 8812d66..97aeabb 100644
--- a/queries/python/highlights.scm
+++ b/queries/python/highlights.scm
@@ -140,7 +140,11 @@

 (comment) @comment
 (string) @string
-(escape_sequence) @string.escape
+[
+  (escape_sequence)
+  "{{"
+  "}}"
+] @string.escape

 ; Tokens
```

**Comment by stsewd** [comment]
@meridionaljet did you run `:TSUpdate`?

**Comment by meridionaljet** [comment]
> @meridionaljet did you run `:TSUpdate`?

Yes, the parser claims to be up to date, and I tried removing and reinstalling. Yet it only works if I check out a commit prior to the aforementioned one.

**Comment by sudo-julia** [comment]
> It's possible that the python parser has been installed by to another location and this is most likely the reason for your issue (that nvim-treesitter cannot update the python parser). Can you the to use `:h nvim_get_runtime` to find `parser/python.so`.
> 
> I meant using the query linter to analyze what's wrong in `queries/python/highlights.scm`

I checked and there are no duplicate python parsers. After checking out `queries/python/highlights.scm` the offending section seems to be exactly what @meridionaljet mentioned: 
![image](https://user-images.githubusercontent.com/62857870/141864778-3e928d02-9701-463a-83ce-4190078fac38.png)

**Comment by sudo-julia** [comment]
Also, here are the problematic vim highlight nodes: 
![image](https://user-images.githubusercontent.com/62857870/141870791-4ffdc05f-6856-4654-8958-f109f3021e8f.png)

**Comment by stsewd** [comment]
that seems like TS isn't installing the latest version of the parsers for some reason, those nodes were added a time ago in the parsers and their commits were updated in our lockfile https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lockfile.json.

Make sure that file matches the one from master in your local installation.

**Comment by sudo-julia** [comment]
The lockfile is the same before and after a reinstall. However, while reinstalling, this error appeared in vim (installing with packer):
```
[packer.nvim] [ERROR 16:55:13] async.lua:20: Error in coroutine: ...ack/packer/start/packer.nvim/lua/packer/plugin_utils.lua:204: Vim(lua):E5108: Error executing lua /usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:335: Overri
ding has-parent?
[packer.nvim] stack traceback:
[packer.nvim] ^I[C]: in function 'error'
[packer.nvim] ^I/usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:335: in function 'add_predicate'
[packer.nvim] ^I...nvim-treesitter/lua/nvim-treesitter/query_predicates.lua:66: in main chunk
[packer.nvim] ^I[C]: in function 'require'
[packer.nvim] ^I...ack/packer/start/nvim-treesitter/lua/nvim-treesitter.lua:13: in main chunk
[packer.nvim] ^I[C]: in function 'require'
[packer.nvim] ^I[string ":lua"]:1: in main chunk
[packer.nvim] ^I[C]: in function 'resume'
[packer.nvim] ^I.../site/pack/packer/start/packer.nvim/lua/packer/async.lua:12: in function <.../site/pack/packer/start/packer.nvim/lua/packer/async.lua:11>
```

Could this be related to the issues I'm having? The `nvim-treesitter` folder is there and everything looks to be in tact, but the issues persist.

**Comment by stsewd** [comment]
Can you try https://github.com/nvim-treesitter/nvim-treesitter/issues/2014#issuecomment-969384614? to see where is the parser being installed.

For the other error, no idea. Can you try with a minimal init.vim or with another package manager?

```vim
call plug#begin('~/.local/share/nvim/plugged')
Plug 'nvim-treesitter/nvim-treesitter'
Plug 'nvim-treesitter/playground'
call plug#end()

lua <<EOF
require'nvim-treesitter.configs'.setup {
  ensure_installed = "all",
  highlight = {
    enable = true
  },
  playground = {
    enable = true,
  },
}
EOF
```

**Comment by sudo-julia** [comment]
This fixed it! `:echo nvim_get_runtime("*/python.so", v:true)` returned two locations: `~/.local/share/nvim/site/parsers/python.so` and `~/.local/share/nvim/pack/packer/start/nvim-treesitter/parser/python.so`. Somehow the first location had eluded my searches with `fd` and `find` (100% user error). Removing the duplicate folder located at `~/.local/share/nvim/site/parsers` solved the issue, as seen in #1887. I apologize for overlooking this and recommend running `:echo nvim_get_runtime("*parser.so", v:true)` for anyone else who's having this issue. Thanks a million times over for the help!
